hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotPurgeRequest.java

```diff
@@ -94,12 +98,16 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
         SnapshotInfo nextSnapshot = SnapshotUtils
             .getNextActiveSnapshot(fromSnapshot,
                 snapshotChainManager, omSnapshotManager);
+
         updateSnapshotInfoAndCache(nextSnapshot, omMetadataManager,
             trxnLogIndex, updatedSnapInfos, true);
+        updateSnapshotChainAndCache(omMetadataManager, fromSnapshot,
```

Comment:
There is a possibility that snapshot chain may get corrupted because of batch operation.
Created jira to revisit it: https://issues.apache.org/jira/browse/HDDS-9198